engine/engine.go

```diff
@@ -146,6 +146,10 @@ func (e *Engine) StartWorkflow(ctx context.Context, name string, context []byte,
 			ids = diff(ids, wRunningIDs) // replace our ids with the set of NON-outstanding ids
 			if len(ids) < 1 {
 				// if all IDs are already running, then return an error
+
+				// log ids
+				logger.Debug(logkeys.Message, fmt.Sprintf("ids: %s, wRIDs: %s", ids, wRunningIDs))
```

Comment:
There's an assumption that ids may be 10's of 1000's long, so we'll not want to log them all. The convention is to log the first one and then a count. But also we've created an explicit error over in 4f62610c3eaae1ce80f716d8cf900ddb43167ffb. If you thought that it was worth it you could add those ids to a custom error struct and handle them upstream however you wish in a higher level. As is though we'll want to drop this log line.